translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,294 @@
+    data_source['update_date'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+    data_source['location_method'] = info['metadata']['wz_location_method']
+    data_source['lrs_type'] = info['metadata']['lrs_type']
+    wzd['road_event_feed_info']['data_sources'] = [data_source]
+
+    wzd['type'] = 'FeatureCollection'
+    sub_identifier = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in
+                             range(6))  # Create random 6 character digit/letter string
+    road_event_id = str(uuid.uuid4())
+    ids = {}
+    ids['sub_identifier'] = sub_identifier
+    ids['road_event_id'] = road_event_id
+
+    wzd['features'] = []
+
+    
+    # Parse alert to WZDx Feature    
+    feature = parse_alert(message, callback_function=unsupported_message_callback)
+    if feature:
+        wzd['features'].append(feature)
+    if not wzd['features']:
+        return None
+    wzd = add_ids(wzd, True)
+    return wzd
+
+
+
+
+
+# function to parse polyline to geometry line string
+def parse_polyline(poly):
+    if not poly or type(poly) != str:
+        return None
+    poly= poly[len('LINESTRING (' ): -1]
+    polyline = poly.split(', ')
+    coordinates = []
+    for i in polyline:
+        coords = i.split(' ')
+        if len(coords) >= 2 and re.match('^-?([0-9]*[.])?[0-9]+$', coords[0]) and re.match('^-?([0-9]*[.])?[0-9]+$', coords[1]):
+            coordinates.append([float(coords[0]), float(coords[1])])
+    return coordinates
+
+
+
+# function to get event status
+def get_event_status(start_time_string, end_time_string):  
+
+    start_time = datetime.fromtimestamp(start_time_string)
+    current_time = datetime.now()
+    future_date_after_2weeks = current_time + \
+                        timedelta(days = 14)
+    event_status = "active"
+    if current_time < start_time:
+        if start_time < future_date_after_2weeks  :
+            event_status = "pending"
+        else:
+            event_status = "planned"
+
+
+    elif end_time_string:
+        end_time = datetime.fromtimestamp(end_time_string)
+        if end_time < current_time:
+            event_status = "completed"
+    return event_status
+
+
+
+# Parse COtrip alert to WZDx
+def parse_alert(alert, callback_function=None):
+    if not validate_alert(alert):
+        if callback_function:    
+            callback_function(alert)
+        return None
+    event = alert.get('event', {})
+    feature = {}
+    geometry = {}
+    geometry['type'] = "LineString"
+    geometry['coordinates'] = parse_polyline(event['geometry'])
+
+    feature['type'] = "Feature"
+    properties = {}
+
+    # road_event_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['road_event_id'] = ''
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # data_source_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['data_source_id'] = ''
+
+    # start_date
```

Comment:
Quite a few of these comments still aren't getting us anything. We can already see that this is setting the 'start_date' value, and don't need the comment. Same for most of the below comments.

Reply:
I like these comments because they break up the code, and make it obvious which code is for what field, especially when a field uses a multi-line expression

Reply:
@payneBrandon  do you still want to remove these comments?